base/pkg/entry.jl

```diff
@@ -197,7 +197,11 @@ function clone(url::AbstractString, pkg::AbstractString)
         end
     catch err
         isdir(pkg) && Base.rm(pkg, recursive=true)
-        rethrow(err)
+        if err.msg == "Unexpected HTTP status code: 404"
```

Comment:
I think you should also check that `isa(err, GitError)`, since not all exceptions have a `msg` field.

I wonder whether there shouldn't even be an error code which could be checked, rather than comparing the whole string. That would be more robust (but it would require more significant changes...).